File: backend/app/data_layer/contracts/conversation.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class ConversationSession:
    session_id: str
    title: str
    created_at: str = ""
    updated_at: str = ""


@dataclass
class ConversationMessage:
    session_id: str
    role: str
    content: str
    created_at: str = ""
    sources_json: str | None = None

File: frontend/src/types/conversation.ts
/** 会话摘要 */
export interface ConversationSession {
  session_id: string
  title: string
  created_at: string
  updated_at: string
}

/** 会话消息（API 返回格式） */
export interface ConversationMessage {
  session_id: string
  role: string
  content: string
  created_at: string
  sources_json: string | null
  blocks_json: string | null
}
